R/raster-element.R

```diff
@@ -0,0 +1,93 @@
+#' \code{\link{add_physical} to generate this physical list and see the 
+#' documentation for \code{\link{add_physical} for more information. 
+#' @param spatial_reference This is the name of a predefined coordinate system. 
+#' The acceptable values for \code{spatial_reference} can be found at 
+#' \href{https://eml.ecoinformatics.org/schema/index.html}{EDI Schema}.
+#' @param horizontal_accuracy  The accuracy of horizontal locational measurements within the data.
+#' @param vertical_accuracy The accuracy of vertical locational measurements within the data. 
+#' @param cell_size_x The width of the cell in the x direction.
+#' @param cell_size_y The width of the cell in the y direction. 
+#' @param number_of_bands The number of bands in the image. 
+#' @param raster_origin  The corner location of the pixel it should have the 
+#' minimum x and y values
+#' @param rows Max number of raster object along the y-axis
+#' @param columns Max number of raster objects along the x-axis
+#' @param verticals Max number of raster objects along the z-axis
+#' @param cell_geometry Geometric representation of the cell's content
+#'
+#' @return A list that contains all the required elements of the spatialRaster 
+#' section of an EML document. 
+#' @export
+#'
+#' @examples
+#' add_raster(file_name = "Rasterfiles.zip" ,
+#'            file_description = "A Raster File",
+#'            attribute_list =  add_attribute(attribute_name = "Date",
+#'                                            attribute_definition = "date data collected",
+#'                                            measurement_scale = "dateTime"),
+#'            physical = add_physical("Rasterfiles.zip"),
+#'            spatial_reference = "NAD_1983_StatePlane_California_I_FIPS_0401",
+#'            horizontal_accuracy = "No Information",
+#'            vertical_accuracy =  "No Information",
+#'            cell_size_x = "25",
+#'            cell_size_y = "25",
+#'            number_of_bands = "10",
+#'            raster_origin = "Upper Left",
+#'            rows = "200",
+#'            columns = "6",
+#'            verticals = "1",
+#'            cell_geometry = "pixel")
+#'            
+add_raster <- function(parent_element, file_name, file_description, attribute_list, physical,
+                       spatial_reference, horizontal_accuracy, vertical_accuracy, 
+                       cell_size_x, cell_size_y, number_of_bands, raster_origin, 
+                       rows, columns, verticals, cell_geometry) {
+  
+  required_arguments <- c("file_name", "file_description", "attribute_list", "physical",
+                          "spatial_reference", "horizontal_accuracy", "vertical_accuracy",
+                          "cell_size_x", "cell_size_y", "number_of_bands", "raster_origin",
+                          "rows", "columns", "verticals", "cell_geometry")
+
+  missing_argument_index <- which(c(missing(file_name), missing(file_description), 
```

Comment:
is `parent_element` not a required argument to this function as well?

Reply:
parent_element is required and if not added it will error but it will not send a custom error element. I did this because it is consistent with how the rest of the functions were written but I can add it as a specific error message.